WalletWasabi.Fluent/ViewModels/AddWallet/WelcomePageViewModel.cs

```diff
@@ -1,59 +1,81 @@
+using System.IO;
 using System.Reactive;
-using System.Reactive.Linq;
+using System.Threading.Tasks;
+using System.Windows.Input;
 using ReactiveUI;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models;
+using WalletWasabi.Fluent.Models.UI;
+using WalletWasabi.Fluent.Models.Wallets;
 using WalletWasabi.Fluent.ViewModels.Dialogs.Base;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
 
 namespace WalletWasabi.Fluent.ViewModels.AddWallet;
 
 [NavigationMetaData(Title = "Welcome")]
 public partial class WelcomePageViewModel : DialogViewModelBase<Unit>
 {
-	private const int NumberOfPages = 2;
-	private readonly AddWalletPageViewModel _addWalletPage;
-	[AutoNotify] private int _selectedIndex;
-	[AutoNotify] private string? _nextLabel;
 	[AutoNotify] private bool _enableNextKey = true;
 
-	private WelcomePageViewModel(AddWalletPageViewModel addWalletPage)
-	{
-		_addWalletPage = addWalletPage;
+	[AutoNotify] private int _itemCount;
+	[AutoNotify] private string? _nextLabel;
+	[AutoNotify] private int _selectedIndex;
 
+	public WelcomePageViewModel(UiContext uiContext)
+	{
 		SetupCancel(enableCancel: false, enableCancelOnEscape: false, enableCancelOnPressed: false);
 
-		SelectedIndex = 0;
-		NextCommand = ReactiveCommand.Create(OnNext);
-		CanGoBack = this.WhenAnyValue(x => x.SelectedIndex, i => i > 0);
+		CanGoBack = this.WhenAnyValue(x => x.SelectedIndex, x => x > 0);
+		CanGoNext = this.WhenAnyValue(x => x.SelectedIndex, x => x.ItemCount, (x, y) => x < y - 1);
+		NextCommand = ReactiveCommand.Create(() => SelectedIndex++, CanGoNext);
 		BackCommand = ReactiveCommand.Create(() => SelectedIndex--, CanGoBack);
 
-		this.WhenAnyValue(x => x.SelectedIndex)
-			.Subscribe(
-				x =>
-				{
-					NextLabel = x < NumberOfPages - 1 ? "Continue" : "Get Started";
-					EnableNextKey = x < NumberOfPages - 1;
-				});
-
-		this.WhenAnyValue(x => x.IsActive)
-			.Skip(1)
-			.Where(x => !x)
-			.Subscribe(x => EnableNextKey = false);
+		CreateWalletCommand = ReactiveCommand.Create(() => uiContext.Navigate().To().WalletNamePage(WalletCreationOption.AddNewWallet));
+		ConnectHardwareWalletCommand = ReactiveCommand.Create(() => uiContext.Navigate().To().WalletNamePage(WalletCreationOption.ConnectToHardwareWallet));
+		ImportWalletCommand = ReactiveCommand.CreateFromTask(OnImportWalletAsync);
+		RecoverWalletCommand = ReactiveCommand.Create(() => uiContext.Navigate().To().WalletNamePage(WalletCreationOption.RecoverWallet));
```

Comment:
These commands shouldn't exist here. When the user went through the dialogs, just navigate to the add wallet page as it was.